packages/flutter/lib/src/material/toggle_buttons.dart

```diff
@@ -667,33 +674,127 @@ class ToggleButtons extends StatelessWidget {
       final BorderSide borderSide = _getBorderSide(index, theme, toggleButtonsTheme);
       final BorderSide trailingBorderSide = _getTrailingBorderSide(index, theme, toggleButtonsTheme);
 
-      return _ToggleButton(
-        selected: isSelected[index],
-        textStyle: textStyle,
-        constraints: constraints,
-        color: color,
-        selectedColor: selectedColor,
-        disabledColor: disabledColor,
-        fillColor: fillColor,
-        focusColor: focusColor ?? toggleButtonsTheme.focusColor,
-        highlightColor: highlightColor ?? toggleButtonsTheme.highlightColor,
-        hoverColor: hoverColor ?? toggleButtonsTheme.hoverColor,
-        splashColor: splashColor ?? toggleButtonsTheme.splashColor,
-        focusNode: focusNodes != null ? focusNodes![index] : null,
-        onPressed: onPressed != null
-          ? () {onPressed!(index);}
-          : null,
-        mouseCursor: mouseCursor,
+      final Set<MaterialState> states = <MaterialState>{
+          if (isSelected[index] && onPressed != null) MaterialState.selected,
+          if (onPressed == null) MaterialState.disabled,
+      };
+      final Color effectiveFillColor = _ResolveFillColor(fillColor
+        ?? toggleButtonsTheme.fillColor).resolve(states)
+        ?? _DefaultFillColor(theme.colorScheme).resolve(states);
+      final Color currentColor;
+      if (onPressed != null && isSelected[index]) {
+        currentColor = selectedColor
+          ?? toggleButtonsTheme.selectedColor
+          ?? theme.colorScheme.primary;
+      } else if (onPressed != null && !isSelected[index]) {
+        currentColor = color
+          ?? toggleButtonsTheme.color
+          ?? theme.colorScheme.onSurface.withOpacity(0.87);
+      } else {
+        currentColor = disabledColor
+          ?? toggleButtonsTheme.disabledColor
+          ?? theme.colorScheme.onSurface.withOpacity(0.38);
+      }
+      final TextStyle currentTextStyle = textStyle
+        ?? toggleButtonsTheme.textStyle
+        ?? theme.textTheme.bodyText2!;
+      final BoxConstraints currentConstraints = constraints
+        ?? toggleButtonsTheme.constraints
+        ?? const BoxConstraints(
+             minWidth: kMinInteractiveDimension,
+             minHeight: kMinInteractiveDimension,
+           );
+
+      final Size minSize;
```

Comment:
Change this to minPaddingSize or some such for clarity
```suggestion
      final Size minPaddingSize;
```